pkg/controllermanager/controller/project/project_control_delete.go

```diff
@@ -44,7 +44,8 @@ func (r *projectReconciler) delete(ctx context.Context, project *gardencorev1bet
 
 		if !isEmpty {
 			r.reportEvent(project, true, gardencorev1beta1.ProjectEventNamespaceNotEmpty, "Cannot release namespace %q because it still contains Shoots.", *namespace)
-			_ = kutil.TryUpdateStatus(ctx, retry.DefaultBackoff, gardenClient, project, func() error {
+
+			_ = kutil.TryPatchStatus(ctx, retry.DefaultBackoff, gardenClient, project, func() error {
```

Comment:
Sorry, `TryPatchStatus` is not what I had in mind. I was rather interested in getting rid of retries, but `TryPatchStatus` also does optimistic locking, so it can also cause conflicts: https://github.com/gardener/gardener/blob/03794c4fab012f16625540cd2989ab8f328be93d/pkg/utils/kubernetes/patch.go#L61

Can you rather do something like this?
https://github.com/gardener/gardener/blob/6ca719c367521bf7e2c1d69f0720af6923a11616/pkg/controllermanager/controller/plant/plant_control.go#L210-L220

Reply:
Okay, thanks, maybe b9789cc is better :)